Lib/unittest/mock.py

```diff
@@ -817,8 +817,17 @@ def _get_call_signature_from_name(self, name):
             if child is None or isinstance(child, _SpecState):
                 break
             else:
+                # If an autospecced object is attached using attach_mock the
+                # child would be a function with mock object as attribute from
+                # which signature has to be derived. If there is no signature
+                # attribute then fallback to None to ensure old signature is
+                # not used.
+                child = _extract_mock(child)
                 children = child._mock_children
-                sig = child._spec_signature
+                try:
+                    sig = child._spec_signature
+                except AttributeError:
```

Comment:
In which situation will we fall here? Can you add a tests that exercises this?

Reply:
I couldn't think of a situation that would get to this as autospecced ones should have this. But given that this was a regression of the previous change without tests catching it I added it as a fallback to have signature as None in case there are scenarios not tested here or try to use `getattr` with None as default.

Reply:
I personally don't like the idea of having code "just if", but you wrote this function initially, so... 🤷‍♂  😛

Reply:
If we can't prove it's needed, let's drop it until we can. The backport caught up now, just want to get as many upstream PRs merged first before releasing, so we should get good feedback if this code can be reached.